tests/unit/s2n_client_hello_test.c

```diff
@@ -646,6 +646,33 @@ int main(int argc, char **argv)
 
             EXPECT_SUCCESS(s2n_config_free(config));
         };
+
+        /* Error if no cipher suites written. */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+
+            struct s2n_cipher_suite cipher_suite = s2n_ecdhe_ecdsa_with_aes_128_gcm_sha256;
+            struct s2n_cipher_suite *cipher_suites = &cipher_suite;
+            struct s2n_cipher_preferences cipher_preferences = {
+                .suites = &cipher_suites,
+                .count = 1,
+            };
+            struct s2n_security_policy policy = *conn->config->security_policy;
+            policy.cipher_preferences = &cipher_preferences;
+            conn->security_policy_override = &policy;
+
+            /* Fails with no cipher suites available / written */
+            cipher_suite.available = false;
+            EXPECT_FAILURE_WITH_ERRNO(s2n_client_hello_send(conn),
+                    S2N_ERR_INVALID_CIPHER_PREFERENCES);
+
+            /* Succeeds with one cipher suite available / written */
+            /* cppcheck-suppress redundantAssignment */
```

Comment:
Hmm, can cppcheck just not follow the chain of pointers? Might be worth including that this is false positive from cppcheck, and not some other reason we are choosing to ignore it.

Reply:
Yup, cppcheck is really bad at this specific kind of indirection. It's not the first time this has come up for our unit tests: https://github.com/search?q=repo%3Aaws%2Fs2n-tls%20redundantAssignment&type=code I haven't figured out a way to fix it other than just suppressing it.

I can still add a comment if you want, but I think it's common enough that the comment would be unnecessary / need to be repeated a lot :)